lib/serialization/serializationOptions.ts

```diff
@@ -76,6 +77,47 @@ export enum SerializationOutputType {
   XML
 }
 
+export function failSerializeTypeCheck(options: SerializationOptions, propertyPath: PropertyPath, value: any, expectedTypeDescription: string): void {
```

Comment:
When we say that the type of a parameter is `any` that means all bets are off w.r.t it's type. It could be a string, boolean, object, etc. or it could be null or undefined. The latter part (null or undefined) makes it optional.

In the above signature type of value is `any` which gives it the notion of it being optional. However, the next parameter after it `expectedTypeDescription: string` is a required parameter. This is an anti-pattern with defining a function where all required parameters should come before optional parameters.

It would be nice to flip the order of last two parameters as follows:
```js
export function failSerializeTypeCheck(options: SerializationOptions, propertyPath: PropertyPath, expectedTypeDescription: string, value: any): void;
```
This makes it easier for  the caller of this method to not provide value if it is not present.

Reply:
On another thought, I dont expect these methods to be called by customers. So we should be fine in this situation. However we should make sure we don't do this for a public facing API.

Reply:
`value` is definitely not optional. `value` could be templatized, but I thought it was pointless because the type wasn't used anywhere. I understand the anti-pattern of optional parameters before required parameters, but does knowing `value` is required change your opinion?

Reply:
value is required but it can still be null or undefined, right?

Reply:
Yup, it can. And those are valid values for the `value` parameter.